src/value/owned.rs

```diff
@@ -340,10 +385,65 @@ impl From<OwnedValue> for OwnedElement {
     }
 }
 
+impl From<IonType> for OwnedElement {
+    fn from(ion_type: IonType) -> Self {
+        OwnedValue::Null(ion_type).into()
+    }
+}
```

Comment:
I don't think I understand this clearly. Did you mean something like this:
```
pub trait Builder {
    type Element: Element + ?Sized;

    /// build an element from Builder
    fn into_elem(self) -> Self::Element;
}
```

where this would then have to use some `enum` for it to return `null`/`clob`/`blob` `Element`. Can you please explain with an example just to be clear on this.

Reply:
Here is a more concrete example definition:

```rust
trait Builder {
    type Element: Element + ?Sized;
    type Sequence: Sequence<Element = Self::Element> + ?Sized;

    fn with_null(e_type: Type) -> Self;
    fn with_clob(seq: Self::Sequence) -> Self;
    fn with_blob(seq: Self::Sequence) -> Self;

    fn into(self) -> Self::Element;
}

trait Element ... {
    type Builder: Builder<Element = Self> + ?Sized;
    ...
}
```

Note that we could flatten this into `Element` (maybe make these methods `new_...` instead of `with...`):

```rust
trait Element ... {
    ...
    new_null(ion_type: IonType) -> Self;
    new_clob(seq: Self::Sequence) -> Self;
    new_blob(seq: Self::Sequence) -> Self;
}
```

Maybe the latter is preferable--I'd love to get your take and @zslayton to weigh in on that.